### ISSUE TITLE
don't fail if h2's "objects.inv" cannot be downloaded

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8823]
|[<img alt="markrwilliams's avatar" src="https://avatars.githubusercontent.com/u/704247?s=50" width="50" height="50">](https://github.com/markrwilliams)| @markrwilliams reported|
|-|-|
|Trac ID|trac#8823|
|Type|defect|
|Created|2016-09-16 13:39:43Z|

Sometimes builds fail because they can't retrieve h2's `objects.inv`.  Twisted should have its own fallback to avoid these failures.

See https://github.com/twisted/twisted/pull/532#issuecomment-247534734

<details><summary>Searchable metadata</summary>

```
trac-id__8823 8823
type__defect defect
reporter__markrwilliams markrwilliams
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1474033183343675 1474033183343675
changetime__1484059911234650 1484059911234650
version__None None
owner__Amber_Brown__hawkowl_____ Amber Brown <hawkowl@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

The alternative approach is that I should mirror h2's `objects.inv` to an S3 bucket of some kind.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

If we did check this in, I have no idea how to tell sphinx to _use_ it.  Any thoughts?

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph commented|
|-|-|

This wouldn't help, because the error is in pydoctor, not sphinx, so it's blocked on https://github.com/twisted/pydoctor/issues/129

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

repurposing this slightly since I think the intent here was to address the failing tests.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

If urlopen can open the URLs, go ahead and build; if not, don't. Should be a pretty straightforward change.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset c57a2a2ef4e476b3c7f82706d996ab81021b1418
```
#!CommitTicketReference repository="" revision="c57a2a2ef4e476b3c7f82706d996ab81021b1418"
Merge 8823-skipfornow: Skip pydoctor building if we can't get intersphinx repos

Author: glyph
Reviewer: hawkowl
Fixes: #8823
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun commented|
|-|-|

It looks like the fix here was to try requesting the intersphinx URLs and skip if there's an error.  But ... won't Sphinx just go and re-request them again?  So this doesn't fix problems arising from transient network errors.  It's only handling cases where the resources have actually moved somewhere else and the URLs are wrong?
